From 3646fb573e3cbfc94de7866103ca0b3e33308216 Mon Sep 17 00:00:00 2001
From: Jeff Moyer <jmoyer@redhat.com>
Date: Thu, 23 Oct 2014 18:46:20 -0400
Subject: [kernel] printk: bump LOG_BUF_SHIFT

Message-id: <x49siied3pf.fsf@segfault.boston.devel.redhat.com>
Patchwork-id: 98329
O-Subject: [rhel7 patch] printk: bump LOG_BUF_SHIFT
Bugzilla: 1156189
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Hi,

Sysrq-t output in our RHEL 7 kernel is just immense (maybe due to
CONFIG_SCHED_DEBUG?).  Go ahead, try it!  Even with few processes
running, I'm seeing overflows of the ring buffer, which makes debugging
problems in this way difficult (you need to reboot the system, passing a
new log_buf_len= parameter, which is not a great option if you have a
hard to hit hang).

So, either we can reduce the amount of logging, or we can bump the
printk buffer.  I've elected to bump the size of the printk buffer.

This resolves bug 1156189.

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/configs/generic/CONFIG_LOG_BUF_SHIFT b/redhat/configs/generic/CONFIG_LOG_BUF_SHIFT
index ca85a83..7ed0134 100644
--- a/redhat/configs/generic/CONFIG_LOG_BUF_SHIFT
+++ b/redhat/configs/generic/CONFIG_LOG_BUF_SHIFT
@@ -1 +1 @@
-CONFIG_LOG_BUF_SHIFT=19
+CONFIG_LOG_BUF_SHIFT=20
-- 
1.7.1